tests/shardddl3/run.sh

```diff
@@ -679,6 +679,691 @@ function DM_103() {
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6,6,6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} drop index idx1, add index idx1(c2,c1), drop index idx2, add index idx2(c4,c3);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9,9,9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop index idx1, add index idx1(c2,c1), drop index idx2, add index idx2(c4,c3);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10,10,10,10,10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11,11,11,11,11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12,12,12,12,12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+function DM_123 {
+    run_case 123 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"" \
+    "clean_table" "pessimistic"
+    run_case 123 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_124_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add index idx1(c1), add index idx2(c2);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop index idx1, drop index idx2;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
```

Comment:
Can we check the lock status and downstream indices here?
So as the tests DM125~127.

Reply:
OK, sounds reasonable.

Reply:
What lock status and downstream indices  indicates the rollback is done right?
When I run `dmctl show-dll-locks`, the output is:
```json
{
    "result": true,
    "msg": "",
    "locks": [
        {
            "ID": "test-`shardddl`.`tb`",
            "task": "test",
            "mode": "optimistic",
            "owner": "",
            "DDLs": [
            ],
            "synced": [
                "mysql-replica-01-`shardddl1`.`tb1`",
                "mysql-replica-02-`shardddl1`.`tb1`",
                "mysql-replica-02-`shardddl1`.`tb2`"
            ],
            "unsynced": [
            ]
        }
    ]
}
```
Should we check if the `unsynced` list is empty?

Reply:
> What lock status and downstream indices indicates the rollback is done right?
> When I run `dmctl show-dll-locks`, the output is:
> 
> ```json
> {
>     "result": true,
>     "msg": "",
>     "locks": [
>         {
>             "ID": "test-`shardddl`.`tb`",
>             "task": "test",
>             "mode": "optimistic",
>             "owner": "",
>             "DDLs": [
>             ],
>             "synced": [
>                 "mysql-replica-01-`shardddl1`.`tb1`",
>                 "mysql-replica-02-`shardddl1`.`tb1`",
>                 "mysql-replica-02-`shardddl1`.`tb2`"
>             ],
>             "unsynced": [
>             ]
>         }
>     ]
> }
> ```
> 
> Should we check if the `unsynced` list is empty?

Yes, I think it's fine ... for now. But in the future, maybe we shouldn't show this lock at all.

Reply:
Due the incorrect behavior of show-dll-locks, we can't checking it by seeing the unsynced list. Do you have any other suggestion?